libheif/plugins/decoder_ffmpeg.cc

```diff
@@ -177,7 +177,11 @@ static struct heif_error ffmpeg_v1_push_data(void* decoder_raw, const void* data
       NalUnit* nal_unit = new NalUnit();
       nal_unit->set_data(cdata + ptr, nal_size);
 
+      NalUnit* old_nal_unit = decoder->NalMap[nal_unit->unit_type()];
```

Comment:
Why not delete it immediately?
You also don't need to test for NULL. `delete` handles this automatically.

Reply:
Makes sense. Will update

Reply:
Fixed in ed27e0d0fa0ea797b94785459787352388a87437